src/ReturnTypeExtension/NodeGetAttributeTypeExtension.php

```diff
@@ -51,18 +51,16 @@ public function getTypeFromMethodCall(
         MethodReflection $methodReflection,
         MethodCall $methodCall,
         Scope $scope
-    ): Type {
-        $returnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
```

Comment:
I think to keep 1 - 1 behaviour, `selectFromArgs()` should be used

```php
$returnType = ParametersAcceptorSelector::selectFromArgs($scope, $methodCall->getArgs(), $methodReflection->getVariants())->getReturnType();
```

Reply:
when returning `null` PHPStan will use:

https://github.com/phpstan/phpstan-src/blob/e95f79b2a3fff1f749cdd6dc9eaab2d2600def1a/src/Analyser/MutatingScope.php#L5616-L5621

Reply:
`getReturnType()` return PHPStan type afaik, it maybe missing test here, but allow return null is changing behavior here

Reply:
see https://phpstan.org/developing-extensions/dynamic-return-type-extensions

<img width="707" alt="grafik" src="https://github.com/user-attachments/assets/5106753c-e56c-4dea-ae16-743f8b862093">

Reply:
If I understand it correctly, it fallbacks to `selectFromArgs()` when `null` is returned.

LGTM :+1:

Reply:
exactly